### PR TITLE
Migrate `RelayLegacyDto` to `zod`

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

This migrates the validation of `RelayLegacyDto` to `zod`.

## Changes

- Remove `RelayLegacyDtoValidationPipe` and associated schemas.
- Add test-covered `RelayLegacyDtoSchema` and infer types from it.
- Propagate type requirements.

### CODE REVIEW

File: src/routes/relay/entities/schemas/relay.legacy.dto.schema.ts [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
-  type: 'object',
-  properties: {
-    chainId: { type: 'string' },
-    to: { type: 'string' },
-    data: { type: 'string' },
-    gasLimit: { oneOf: [{ type: 'string' }, { type: 'null', nullable: true }] },
-  },
-  required: ['chainId', 'to', 'data'],
-};
+export const RelayLegacyDtoSchema = z.object({
```
Comment (fmrsabino) [inline src/routes/relay/entities/schemas/relay.legacy.dto.schema.ts:6]: nitpick: to move the schema closer to the entity. In this case to `src/routes/relay/entities/relay.legacy.dto.entity.ts` (I think it helps with code navigation – less jumps).

**Review by fmrsabino — APPROVED** [review]
🚀

File: src/routes/relay/entities/schemas/relay.legacy.dto.schema.ts (hunk 1 above)
Comment (iamacook) [reply src/routes/relay/entities/schemas/relay.legacy.dto.schema.ts]: Moved in 08933097cdc0dcb961862e1daabdd681dd331b20.

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**